### ISSUE TITLE
Get compound @@id value from query for joint tables

### ISSUE THREAD

**Issue by devklepacki** [prisma/prisma#17467]
## Problem

When querying data from joint tables there may be no `id` field when using `compound ids`.

Maybe there's a way but I haven't found it. Also tried with `@@id([typeId, spaceId], name: "id")` or `@@id([typeId, spaceId], map: "id")` (the latter is not working at all due to using `mysql` provider) but with no luck. It seems like the `map` field could be the answer but I cannot use it.

In some cases getting this composite id value would be useful when ex. generating JSX with `.map()` that requires a `key` attribute.

```prisma
model Space {
  ...
  classes Class[]
  spots   SpaceTypeSpots[]
  ...
}

model SpaceTypeSpots {
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  type   ClassType @relation(fields: [typeId], references: [id], onDelete: Cascade, onUpdate: Restrict)
  typeId String

  space   Space  @relation(fields: [spaceId], references: [id], onDelete: Cascade, onUpdate: Restrict)
  spaceId String

  spots   Int
  enforce Boolean @default(false)

  @@id([typeId, spaceId])
  @@index([typeId])
  @@index([spaceId])
  @@map("spots")
}
```

```tsx
const space = await ctx.prisma.space.findUnique({
  where: {
    id: input.id,
  },
  include: {
    classes: { ... },
    spots: { ... },
  },
})
```

```jsx
<ul>
  {space?.spots.map((spot) => (
    <li key={spot.id}></li> // id doesn't exist
  ))}
</ul>
```

For now I just do:
```tsx
const spotsWithId = space?.spots.map((spot) => ({
  ...spot,
  id: `${spot.typeId}_${spot.spaceId}`,
}))
```
and then use it in JSX, but it'd be great to just get this value from Prisma query itself.

## Suggested solution

```tsx
const space = await ctx.prisma.space.findUnique({
  where: {
    id: input.id,
  },
  include: {
    classes: { ... },
    spots: {
      compound_id: true, // <-- get compound Id value?
      include: { ... },
    },
  },
})
```

**Comment by janpio** [comment]
The `@@id` is not a value by itself, it is a primary key using two fields. As Prisma needs to be able to query via that primary key, we have a "generated" name for it that can be used in `where`: `typeId_spaceId` in your case. So when you do a `findUnique` on `SpaceTypeSpots` you will have that field available in `where` and can provide an object with 2 values to it - that way Prisma knows that this is the "unique" field (which `@@id` also are) that guarantees uniqueness.

What would be the usecase for also returning this `typeId_spaceId` in a query?

**Comment by SevInf** [comment]
Yes, as Jan mentioned "compound id value" is not an actual thing that exists anywhere in Prisma or database and can be returned. 
You can pretty easily add one yourself using [`result` extensions](https://www.prisma.io/docs/concepts/components/prisma-client/client-extensions/result) and it will be available everywhere your model is used.

**Comment by devklepacki** [comment]
It seems like the `result extension` is exactly what was needed here 🙂 Thank you, I wasn't aware of such a thing.

Although adding it for each computed field is necessary we can say it's a minor annoyance:
```jsx
result: {
      userSchoolRole: {
        compoundId: {
          needs: { userId: true, schoolId: true },
          compute(data) {
            return `${data.userId}_${data.schoolId}`
          },
        },
      },
      spaceTypeSpots: {
        compoundId: {
          needs: { spaceId: true, typeId: true },
          compute(data) {
            return `${data.spaceId}_${data.typeId}`
          },
        },
      },
      // ... and so on
    },
```
Just having it as a `compound_id: true` field available in a query where `@@id` is used would be much easier, but we can't have everything, don't we?
